src/Repository/UserRepository.php

```diff
@@ -56,6 +56,19 @@ public function get($id): ?object
         });
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function delete($id)
+    {
+        if ($id == null) {
+            return null;
+        }
+
+        $this->mgmtApi->users()->delete($id);
```

Comment:
I think it would be good to check the response status and throw an error if it's not successful (HTTP status 204).